lib/components/flare_component.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:ui';
+
+import 'component.dart';
+import 'package:flame/flare_animation.dart';
+
+class FlareComponent extends PositionComponent {
+  FlareAnimation _flareAnimation;
+
+  FlareComponent(
+      String fileName, String animation, double width, double height) {
+    FlareAnimation.load(fileName).then((loadedFlareAnimation) {
+      _flareAnimation = loadedFlareAnimation;
+
+      _flareAnimation.updateAnimation(animation);
+      _flareAnimation.width = width;
+      _flareAnimation.height = height;
+    });
+  }
+
+  @override
+  bool loaded() => _flareAnimation != null;
+
+  @override
+  void render(Canvas canvas) {
+    _flareAnimation.render(canvas);
+  }
+
+  @override
+  void update(double dt) {
+    _flareAnimation.x = x;
```

Comment:
this will give a null pointer if _flareAnimation is still null, AFAICT. could you please verify that?

Reply:
Only if the user calls the update manually, if handled by the BaseGame class, no NPE will happen because the loaded condition is that `_flareAnimation` is not null.

Reply:
I strongly disagree, AFAIR the loaded property is only checked upon rendering, not upon updating. All components are always updated.

Reply:
cf. https://github.com/luanpotter/flame/blob/master/lib/game.dart#L142

Reply:
But make a test to see which is the case, please, I might be incorrect

Reply:
Yeah, I added the `if` to verify if the instance is null.